scenes/objects/wall_objects/wall_attachment_component.gd

```diff
@@ -0,0 +1,642 @@
+
+## Whether this object should automatically setup wall attachment on ready
+@export var auto_setup_attachment: bool = true
+
+## Distance threshold for magnetic pull effect (larger zone for attraction)
+@export var magnetic_pull_distance: float = 0.3
+
+## Distance threshold for actual re-attachment (smaller zone for precision)
+@export var reattachment_distance: float = 0.05
+
+var orientation_threshold = 0.5  # DEBUG: This should be > 0.0 to prevent front-facing attachment!
+
+## Strength of magnetic pull (higher = stronger pull)
+@export var magnetic_pull_strength: float = 4.0  # Increased for stronger attraction
+
+## Speed of smooth lerp movement when snapping
+@export var snap_lerp_speed: float = 12.0  # Increased for faster snapping
+
+
+#--- private variables - order: export > normal var > onready -------------------------------------
+
+## Reference to the wall mount (StaticBody3D)
+var wall_mount: StaticBody3D = null
+
+## Joint connecting this object to the wall mount
+var attachment_joint: Joint3D = null
+
+## Whether this object is currently attached to a wall
+var is_attached_to_wall: bool = false
+
+## Original rotation when attached to wall (for magnetic orientation correction)
+var original_wall_rotation: Quaternion = Quaternion.IDENTITY
+
+## Timer for checking re-attachment proximity
+var reattachment_timer: Timer = null
+
+## Timer for monitoring breakaway forces
+var breakaway_monitor_timer: Timer = null
+
+## Whether magnetic pull is currently active
+var is_magnetic_pull_active: bool = false
+
+## Original material of the parent object (for restoring after feedback)
+var original_materials: Array[Material] = []
+
+## MeshInstance3D nodes for applying visual feedback
+var mesh_instances: Array[MeshInstance3D] = []
+
+### Built-in Virtual Overrides --------------------------------------------------------------------
+
+func _ready():
+	# Ensure we're attached to a PhysicsBody3D
+	if not (get_parent() is PhysicsBody3D):
+		push_error("WallAttachmentComponent must be child of a PhysicsBody3D")
+		return
+	
+	var parent_body = get_parent() as PhysicsBody3D
+	
+	# Cache mesh instances for visual feedback
+	_cache_mesh_instances()
+	
+	if auto_setup_attachment and wall_direction != -1:
+		call_deferred("_setup_wall_attachment")
+	
+	# Connect to physics signals for breakaway detection if parent supports them
+	if parent_body.has_signal("body_entered"):
+		parent_body.body_entered.connect(_on_body_entered)
+	if parent_body.has_signal("sleeping_state_changed"):
+		parent_body.sleeping_state_changed.connect(_on_sleeping_state_changed)
+
+
+func _physics_process(delta: float):
+	# Handle magnetic pull when detached
+	if not is_attached_to_wall and _is_in_magnetic_range() and wall_mount:
+		# DEBUG: Check if parent is being grabbed by player
+		var parent_body = get_parent() as PhysicsBody3D
+		var is_being_grabbed = false
+		if parent_body and parent_body.has_method("get_attachment_status"):
```

Comment:
Avoid string based checks (prefer type checks when possible), and specially avoid doing them in _process/_physics_process or other performance critical code paths.